python/thunder/rdds/series.py

```diff
@@ -883,6 +885,229 @@ def toSpatialSeries(self):
+        values will be retained for the returned Series. If filtering, this is inverted and only 
+        those elements will be ddropped.
+
+        Parameters:
+        -----------
+        level: An integer or an list of integers specificying which levels in the multilevel index to use when
+            performing the selection.
+        
+        val: A list of lists specifying which of the values at each of the specified levels will be selected.
+            If a single level is specified, then this can be a single integer as well.
+
+        squeeze: If True, then any level where only a single value is specified will be dropped in the resulting
+            index. Useful when one wants to maintain unique indices.
+
+        filter: If True, then all elemented EXCPET those signified by 'level' and 'value' will be selected.
+        """
+
+        try:
+            level[0]
+        except:
+            level = [level]
+        try:
+            val[0]
+        except:
+            val = [val]
+        
+        remove = []
+        if len(level) == 1:
+            try:
+                val[0][0]
+            except:
+                val = [val]
+            if squeeze and not filter and len(val) == 1:
+                remove.append(level[0])
+        else:
+            for i in xrange(len(val)):
+                try:
+                    val[i][0]
+                except:
+                    val[i] = [val[i]]
+                if squeeze and not filter and len(val[i]) == 1:
+                    remove.append(level[i])
+                                
+        if len(level) != len(val):
+            raise ValueError("list of levels must be of same length as list of corresponding values")
+
+        p = product(*val)
+        selected = set([x for x in p])
+
+        #TODO: this could be more efficient if _makeMasks also accepted the desired values so that
+        #we not produce ALL possible masks which we must then filter down to the ones we want
+        masks, ind = self._makeMasks(index=self.index, level=level)
+        nmasks = len(masks)
+        masks = array([masks[x] for x in xrange(nmasks) if tuple(ind[x]) in selected])
+
+        finalMask = masks.any(axis=0)
+        if filter:
+            finalMask = logical_not(finalMask)
+        bcMask = self.rdd.ctx.broadcast(finalMask)
+        
+        rdd = self.rdd.mapValues(lambda v: v[bcMask.value])
+        indFinal = array(self.index)
+        if len(indFinal.shape) == 1:
+            indFinal = array(indFinal, ndmin=2).T
+        indFinal = indFinal[finalMask]
+
+        if squeeze:
+            indFinal = delete(indFinal, remove, axis=1)
+
+        if len(indFinal[0]) == 1:
+            indFinal = ravel(indFinal)
+
+        elif len(indFinal[1]) == 0:
+            indFinal = arange(sum(finalMask))
+
+        return Series(rdd, index=indFinal).__finalize__(self, noPropagate=('_index',))
```

Comment:
The `noPropagate` shouldn't be necessary if an index is being passed during construction, but double check that